connectors/sources/mssql.py

```diff
@@ -62,6 +62,244 @@ def all_schemas(self):
+            )
+            return cursor
+        except Exception as exception:
+            self._logger.warning(
+                f"Something went wrong while executing query. Exception: {exception}"
+            )
+            raise
+
+    async def ping(self):
+        return await anext(
+            self._execute_query(
+                query=self.queries.ping(),
+            )
+        )
+
+    async def get_tables_to_fetch(self):
+        tables = configured_tables(self.tables)
+        return list(
+            map(
+                lambda table: table[0],  # type: ignore
+                await anext(
+                    self._execute_query(
+                        query=self.queries.all_tables(
+                            database=self.database,
+                            schema=self.schema,
+                        )
+                    )
+                ),
+            )
+            if is_wildcard(tables)
+            else tables
+        )
+
+    async def get_table_row_count(self, table):
+        [[row_count]] = await anext(
+            self._execute_query(
+                query=self.queries.table_data_count(
+                    schema=self.schema,
+                    table=table,
+                ),
+            )
+        )
+        return row_count
+
+    async def get_table_primary_key(self, table):
+        return await anext(
+            self._execute_query(
+                query=self.queries.table_primary_key(
+                    schema=self.schema,
+                    table=table,
+                ),
+            )
+        )
+
+    async def get_table_last_update_time(self, table):
+        return await anext(
+            self._execute_query(
+                query=self.queries.table_last_update_time(
+                    schema=self.schema,
+                    table=table,
+                ),
+            )
+        )
+
+    async def data_streamer(self, table):
+        """Streaming data from a table
+
+        Args:
+            table (str): Table.
+
+        Raises:
+            exception: Raise an exception after retrieving
+
+        Yields:
+            list: It will first yield the column names, then data in each row
+        """
+        async for data in self._execute_query(
+            query=self.queries.table_data(
+                schema=self.schema,
+                table=table,
+            ),
+            fetch_many=True,
+            schema=self.schema,
+            table=table,
+        ):
+            yield data
+
+    async def _execute_query(self, query, fetch_many=False, **kwargs):
+        """Executes a query and yield rows
+
+        Args:
+            query (str): Query.
+            fetch_many (bool): Flag to use fetchmany method. Defaults to False.
+
+        Raises:
+            exception: Raise an exception after retrieving
+
+        Yields:
+            list: Column names and query response
+        """
+        retry = 1
+        yield_once = True
```

Comment:
Should this be removed? Function should explicitly return or yield, not change behaviour on a flag.

Reply:
The name `yield_once` is misleading. It means only yield column names first, then rows.

I don't want to make changes here, because once #1401 is merged, this PR will rebase and remove this function.